src/function/table/sniff_csv.cpp

```diff
@@ -0,0 +1,222 @@
+#include "duckdb/function/built_in_functions.hpp"
+#include "duckdb/execution/operator/scan/csv/csv_reader_options.hpp"
+#include "duckdb/common/types/data_chunk.hpp"
+#include "duckdb/execution/operator/scan/csv/csv_sniffer.hpp"
+#include "duckdb/execution/operator/scan/csv/csv_buffer_manager.hpp"
+#include "duckdb/function/table_function.hpp"
+#include "duckdb/main/client_context.hpp"
+#include "duckdb/function/table/range.hpp"
+#include "duckdb/execution/operator/scan/csv/base_csv_reader.hpp"
+#include "duckdb/execution/operator/scan/csv/csv_file_handle.hpp"
+
+namespace duckdb {
+
+struct CSVSniffFunctionData : public TableFunctionData {
+	CSVSniffFunctionData() {
+	}
+	string path;
+	int64_t sample_size = 20480;
+};
+
+struct CSVSniffGlobalState : public GlobalTableFunctionState {
+	CSVSniffGlobalState() {
+	}
+	bool done = false;
+};
+
+static unique_ptr<GlobalTableFunctionState> CSVSniffInitGlobal(ClientContext &context, TableFunctionInitInput &input) {
+	return make_uniq<CSVSniffGlobalState>();
+}
+
+static unique_ptr<FunctionData> CSVSniffBind(ClientContext &context, TableFunctionBindInput &input,
+                                             vector<LogicalType> &return_types, vector<string> &names) {
+	auto result = make_uniq<CSVSniffFunctionData>();
+	result->path = input.inputs[0].ToString();
```

Comment:
Might not be that simple/important - but perhaps it makes sense to allow all parameters that we allow for `read_csv_auto` to be specified here as well, e.g. allowing the user to override the delimiter, etc?